### PR TITLE
New Recipe: ngspice v32.2.0

### PR DESCRIPTION
**Pull request by Keno**
Requires four minor patches, but otherwise builds nicely.
I've submitted the first of these four patches upstream at
https://sourceforge.net/p/ngspice/patches/98/. We'll see
if there's any movement there - if so, I'll submit the other
ones.

### CODE REVIEW

File: N/ngspice/bundled/patches/drop_ac_malloc.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
+commit 8960318bdde32ba8ea38e268fea8ba130b2ae3ba
+Author: Keno Fischer <keno@juliacomputing.com>
+Date:   Mon Oct 5 03:28:56 2020 -0400
+
+    Drop AC_FUNC_*ALLOC configure macros
```
Comment (giordano) [inline N/ngspice/bundled/patches/drop_ac_malloc.patch:5]: I can't find anything directly referencing these macros in any other builders, but if I remember correctly I also wrestled with these macros before, they're just painful.  I also found other people online just getting rid of them to make things work.

Comment (Keno) [reply N/ngspice/bundled/patches/drop_ac_malloc.patch:5]: packages that use them are supposed to ship the rpl_ versions of these functions, but they basically never do

**Comment by giordano** [comment]
https://sourceforge.net/p/ngspice/ngspice/merge-requests/8/ seems relevant for FreeBSD

**Comment by Keno** [comment]
Yeah, freebsd build is fixed on upstream master. I'll bump the base commit.

**Comment by giordano** [comment]
And that broke Windows, beautiful :joy:

**Comment by Keno** [comment]
Yeah, I'm on it.

**Comment by Keno** [comment]
:green_apple: :green_book: :green_circle: :green_heart: :green_salad:

**Comment by Keno** [comment]
Actually, I think I want the shared library also. Let me turn that on.

**Comment by Keno** [comment]
Looks like ngspice is one of those projects that can only build either the executable or the library, so I'm just building everything twice now. Wastes a bit of disk space, but I think it's fine for the convenience. If this somehow becomes the most used jll in the world, we can always consider splitting it.

**Comment by Keno** [comment]
Turned off xspice for now. It was failing in its custom compiler step on windows. Don't feel like debugging that, and I don't think I need it.

**Comment by Keno** [comment]
I'll call this good enough for now.

**Pull request merged**